### PR TITLE
Unify context menu call paths

### PR DESCRIPTION
**Pull request by Outcue**
I found another code path that opens a menu (deploy, configurations) so I have merged the code paths. There is some slinging of shell and visuals as parameters probably because I still am figuring out Window, HopscotchWindow, HopscothShell. etc.

Menu sizing still needs work.

### CODE REVIEW

File: HopscotchForHTML5.ns [hunk 1]
```diff
@@ -2943,14 +2861,55 @@ computeContentForMenu: menuSupplier <[Menu]> ^ <Alien[Div]> = (
...
+	w:: w + edgePad.
+	menu at: 'width' put: w.
+
+	(* Determine optimal placement *)
+    maxLeft:: (window at: 'innerWidth') - w - rightInset.
+    x > maxLeft 
+        ifTrue: [ left:: maxLeft asString ]
+        ifFalse: [ left:: x asString ].
+
+    y < 0
```
Comment (gbracha) [inline HopscotchForHTML5.ns:2900]: If I was being picky, I'd prefer

top:: y < 0 ifTrue: [0] ifFalse:[y as String]

which also makes it obvious that something weird is going on - top is sometimes a Float, as declared, and sometimes a String?

Comment (gbracha) [reply HopscotchForHTML5.ns:2900]: Same with left.

Comment (Outcue) [reply HopscotchForHTML5.ns:2900]: Let me revisit that code. Is the value coming from the dictionary a string that then gets coerced into a float? I don't know the magical internals of your language and how it deals with values coming out of Javascript dicts.

Comment (gbracha) [reply HopscotchForHTML5.ns:2900]: We don't do magical coercions. JS strings stay strings, and floats stay floats. It's possible that JS will happily deal with either. But if we are assigning to top, we should stick to one or the other and agree with the type annotation (the optional typechecker is on the back burner - once the IDE is complete, maybe I'll get back to that).

Comment (Outcue) [reply HopscotchForHTML5.ns:2900]: I am converting it to a string as the value is then prepended before 'px' when setting the style values here:

 (menu at: 'style') 
        at: 'position' put: 'absolute';
        at: 'left' put: left,'px';
        at: 'top' put: top,'px';
        at: 'width' put: w asString,'px'.

Comment (Outcue) [reply HopscotchForHTML5.ns:2900]: I am running into some type conversion issue with this code:

```
(* Determine optimal placement *)
    maxLeft:: (window at: 'innerWidth') - w - rightInset.
	left:: x < maxLeft ifTrue: maxLeft ifFalse:x
	top:: y < 0 ifTrue: [0] ifFalse:[y]
    (menu at: 'style') 
        at: 'position' put: 'absolute';
        at: 'left' put: left,'px';
        at: 'top' put: top,'px';
        at: 'width' put: w asString,'px'.
```

Results in: 

Exception in turn without resolver
Unhandled exception: 
MessageNotUnderstood: SmallInteger value
SmallInteger(_Object) doesNotUnderstand: #value
False ifTrue:ifFalse:

This seems to happen with values that ared retrieved from JS dicts like this:
(window at: 'innerWidth')

Not sure what is going on here.

Comment (gbracha) [reply HopscotchForHTML5.ns]: I think I see the issue. Unlike Smalltalk, Newspeak ifTrue: and friends expect blocks as arguments. Do change it to:

left:: x < maxLeft ifTrue: [maxLeft] ifFalse: [x]

File: HopscotchForHTML5.ns [hunk 2]
```diff
@@ -2893,14 +2893,9 @@ positionMenu: menu <Alien[Div]> forShell: s <HopscotchShell> = (
...
-    x > maxLeft 
-        ifTrue: [ left:: maxLeft asString ]
-        ifFalse: [ left:: x asString ].
-
-    y < 0 
-        ifTrue: [ top:: 0 ]
-        ifFalse: [ top:: y asString ].
-
+	maxLeft out.
+	left:: x < maxLeft ifTrue: maxLeft ifFalse:x
```
Comment (gbracha) [inline HopscotchForHTML5.ns:2897]: Same issue here.

File: HopscotchForHTML5.ns (hunk 1 above)
Comment (Outcue) [reply HopscotchForHTML5.ns]: Argh. The [] and the missing period at the end of the line always get me.

**Pull request merged**